### PR TITLE
Finish lightweight mDNS querier

### PR DESCRIPTION
**Pull request by samkearney**
The querier implements the behavior required by RFC 6762 and RFC 6763, except that it does not currently requery records when their TTL is expiring. That is tracked in https://jira.etcconnect.com/browse/RDMNET-169.

### CODE REVIEW

File: src/rdmnet/disc/lightweight/lwmdns_common.c [hunk 1]
```diff
@@ -150,48 +180,478 @@ const uint8_t* lwmdns_parse_resource_record(const uint8_t*     buf_begin,
...
+  if (binary_atoi(item->value, item->value_len) != E133_DNSSD_TXTVERS)
+    return false;
+
+  return true;
+}
+
+bool parse_txt_item(const TxtRecordItemRef* item, DiscoveredBroker* db, txt_keys_found_mask_t* found_mask)
+{
+  // E133Scope
+  if (item->key_len == sizeof(E133_TXT_SCOPE_KEY) - 1 &&
```
Comment (nwagner) [inline src/rdmnet/disc/lightweight/lwmdns_common.c:436]: I'm seeing a lot of this pattern here, and in the previous function.  Perhaps a utility function that takes two keys & and a size?

File: src/rdmnet/disc/lightweight/lwmdns_common.c [hunk 2]
```diff
@@ -150,48 +180,478 @@ const uint8_t* lwmdns_parse_resource_record(const uint8_t*     buf_begin,
...
+  }
+  return false;
+}
+
+bool parse_manufacturer_item(const TxtRecordItemRef* item, DiscoveredBroker* db, txt_keys_found_mask_t* found_mask)
+{
+  if (item->value_len > 0 && item->value_len < E133_MANUFACTURER_STRING_PADDED_LENGTH)
+  {
+    *found_mask |= TXT_KEY_MANUF_FOUND_MASK;
+    if (strlen(db->manufacturer) != item->value_len || memcmp(db->manufacturer, item->value, item->value_len) != 0)
```
Comment (nwagner) [inline src/rdmnet/disc/lightweight/lwmdns_common.c:572]: Utility func?  It's used in the previous function as well.

File: src/rdmnet/disc/lightweight/lwmdns_common.c [hunk 3]
```diff
@@ -150,48 +180,478 @@ const uint8_t* lwmdns_parse_resource_record(const uint8_t*     buf_begin,
...
+  label->label = next_length_offset + 1;
+  return true;
+}
+
+bool is_rdmnet_service_type_and_domain(const uint8_t* buf_begin, DomainNameLabel* last_non_service_label)
+{
+  DomainNameLabel label = *last_non_service_label;
+
+  // Compare the service type (e.g. _rdmnet)
+  if (!get_domain_name_label(buf_begin, NULL, &label) || (label.length != (sizeof("_rdmnet") - 1)) ||
```
Comment (nwagner) [inline src/rdmnet/disc/lightweight/lwmdns_common.c:633]: Same here.  If I'm being too copy/paste averse, let me know. :)

**Comment by samkearney** [comment]
@nwagner I added all of the utility functions you pointed out.

**Pull request merged**